src/golang/lib/workflow/artifact/artifact.go

```diff
@@ -25,6 +25,12 @@ type Artifact interface {
 	Signature() uuid.UUID
 	Type() shared.ArtifactType
 	Name() string
+	ShouldPersistContent() bool
+
+	// DeleteContent removes the artifact content from storage.
```

Comment:
Can you note that this only deletes the latest artifact result? Also, what happens if the artifact content was never written due to workflow failure.